test/distributed/test_c10d_nccl.py

```diff
@@ -1170,6 +1170,33 @@ def abortpg():
 
             thread.join()
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
+    def test_close_pg(self):
+        # Disable ASYNC_ERROR_HANDLING for this test to ensure we can programmatically
+        # abort the process group.
+        os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "0"
+
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = self._create_process_group_nccl(store, self.opts())
+        device = self.rank_to_GPU[self.rank][0]
+
+        t = torch.rand(10, 10, device=device)
+        # First allreduce to initialize state.
+        pg.allreduce(t)
+
+        # Destroy pg and validate pg is still in working condition since we hold a
+        # reference above.
+        dist.destroy_process_group()
+        pg.allreduce([t])
```

Comment:
Is it possible to call allreduce even after pg destroy? The first allreduce is blocking right?

Reply:
Yes because `destroy_process_group` just removes the PG from the map in `distributed_c10d.py`, but doesn't trigger the destructor of ProcessGroupNCCL since we are holding a reference to the PG object here.

Reply:
Oh I see. That's a good idea to put a test case. Thanks!